ttnn/cpp/ttnn/operations/data_movement/unsqueeze/unsqueeze.cpp

```diff
@@ -12,14 +12,13 @@ ttnn::Tensor UnsqueezeOperation::invoke(const ttnn::Tensor& input_tensor, const
     const auto rank = tensor_shape.rank();
     SmallVector<uint32_t> output_shape_vector;
 
-    TT_FATAL(
-        input_tensor.get_layout() == Layout::ROW_MAJOR or (!tensor_shape.has_tile_padding()),
-        "Currently supporing ROW-MAJOR tensors or TILE tensors with no padding");
-
-    int normal_dim = dim;
+    int normal_dim;
     // Handle negative dimension by converting it to positive
+    TT_FATAL((dim < -rank - 1) || (dim > rank), "Dimension out of range");
     if (dim < 0) {
-        normal_dim += rank + 1;
+        normal_dim = rank + 1 + dim;
```

Comment:
If rank = 4 and dim = -1, normal_dim = 4 + 1 - 1 = 4

Is that intended?

Shouldn't that be 3?

Reply:
While strange this does match the behaviour in torch, where negative dims are applied to dim + rank + 1

The idea is -1 corresponds to the highest value possible, which in this case would be 4

https://pytorch.org/docs/stable/generated/torch.unsqueeze.html